src/averageAges.js

```diff
@@ -37,7 +50,17 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const woman = people.filter(person => person.sex === 'f');
+
+  const allMotherPeople = people.map(person => person.mother);
+
+  const mothers = woman.filter(person => allMotherPeople.includes(person.name));
+
+  const averageAge = getAverageAge(mothers);
+
+  const averageAgeWoman = getAverageAge(woman);
```

Comment:
You don't need to calculate average age of mothers if argument withChildren is not specified. Move logic that relates to this inside if statement and check if withChildren is true.